File: proxy/nodejs/src/utils/keystore.js
const crypto = require('crypto');
const fs = require('fs');
const path = require('path');
const os = require('os');

const KEY_DIR  = path.join(os.homedir(), '.uxermanager');
const KEY_FILE = path.join(KEY_DIR, 'key');

function loadOrCreateKey() {
  if (!fs.existsSync(KEY_DIR)) fs.mkdirSync(KEY_DIR, { recursive: true });
  if (fs.existsSync(KEY_FILE)) {
    return Buffer.from(fs.readFileSync(KEY_FILE, 'utf8').trim(), 'hex');
  }
  const key = crypto.randomBytes(32);
  fs.writeFileSync(KEY_FILE, key.toString('hex'), { encoding: 'utf8', mode: 0o600 });
  return key;
}

module.exports = { loadOrCreateKey };

File: proxy/python/utils/keystore.py
import os
import secrets
from pathlib import Path

KEY_DIR = Path.home() / ".uxermanager"
KEY_FILE = KEY_DIR / "key"


def load_or_create_key() -> bytes:
    KEY_DIR.mkdir(parents=True, exist_ok=True)
    if KEY_FILE.exists():
        hex_key = KEY_FILE.read_text(encoding="utf-8").strip()
        return bytes.fromhex(hex_key)
    key = secrets.token_bytes(32)
    KEY_FILE.write_text(key.hex(), encoding="utf-8")
    try:
        KEY_FILE.chmod(0o600)
    except Exception:
        pass
    return key
